airbyte-integrations/connectors/source-amazon-ads/source_amazon_ads/streams/attribution_report.py

```diff
@@ -70,40 +72,51 @@ class AttributionReport(AmazonAdsStream):
 
     def __init__(self, config: Mapping[str, Any], *args, **kwargs):
         self._start_date = config.get("start_date")
-        self._req_start_date = ""
-        self._req_end_date = ""
-
         super().__init__(config, *args, **kwargs)
 
-    def _set_dates(self, profile: Profile):
-        new_start_date = pendulum.now(tz=profile.timezone).subtract(days=1).date()
-        new_end_date = pendulum.now(tz=profile.timezone).date()
-
-        if self._start_date:
-            new_start_date = max(self._start_date, new_end_date.subtract(days=self.REPORTING_PERIOD))
-
-        self._req_start_date = new_start_date.format(self.REPORT_DATE_FORMAT)
-        self._req_end_date = new_end_date.format(self.REPORT_DATE_FORMAT)
-
     @property
     def http_method(self) -> str:
         return "POST"
 
-    def read_records(self, *args, **kvargs) -> Iterable[Mapping[str, Any]]:
-        """
-        Iterate through self._profiles list and send read all records for each profile.
-        """
+    def path(self, **kvargs) -> str:
+        return "/attribution/report"
+
+    def stream_slices(
```

Comment:
do you test new functionality with unit tests?

Reply:
I have added unit_test for slicing